incubator/hnc/internal/validators/object.go

```diff
@@ -275,16 +280,22 @@ func (o *Object) handleInherited(op k8sadm.Operation, newSource, oldSource strin
 	return deny(metav1.StatusReasonInternalError, "unknown operation: "+string(op))
 }
 
-// isDeletingNS returns true if the namespace of the object is already being deleted, based on the
-// in-memory forest.
-func (o *Object) isDeletingNS(inst *unstructured.Unstructured) bool {
-	o.Forest.Lock()
-	defer o.Forest.Unlock()
-	ns := o.Forest.Get(inst.GetNamespace())
-	if ns == nil {
-		return false
+// validateDeletingNS validates if the namespace of the object is already being deleted
+func (o *Object) isDeletingNS(ctx context.Context, ns string) (bool, error) {
+	nsObj := &corev1.Namespace{}
+	nnm := types.NamespacedName{Name: ns}
+	if err := o.client.Get(ctx, nnm, nsObj); err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
Maybe comment: this should never happen, but if for some bizarre reason the namespace appears to be deleted before the object, we should allow the object to be deleted too.

Perhaps logging something would be nice here as well, but I don't feel too strongly about that.

Reply:
comment added